src/script.ts

```diff
@@ -113,6 +113,17 @@ function enableNightMode(): void {
     document.body.classList.add("nightmode");
 }
 
+function updateClock(): void {
+    const time = new Date();
+    const clockTextElm = document.getElementById("clock-text");
+    if (clockTextElm == null) {
+        console.log("clock text element not found");
+    } else {
+        clockTextElm.textContent = time.toLocaleString("en-US", { hour: "numeric", minute: "numeric", hour12: true });
+    }
+    setTimeout(updateClock, (60 - time.getSeconds()) * 1000);
```

Comment:
The old way of updating this (with `setInterval`, outside of the function body) was better, it keeps the purpose of this function simpler.

I also still think updating this every second is fine but I won't die on that hill :P